### PR TITLE
Add icon to organizations

### PR DESCRIPTION
**Pull request by scarfdestroyer**
I added an `add_organization_icon_worker` as a start. Should these workers (add_project_icon_worker) be combined some how by passing in the model or just left separate?

Also, `organizations` do not currently have a `update` action in the controller, should this be tackled in a different issue?

### CODE REVIEW

File: spec/models/organization_spec.rb [hunk 1]
```diff
@@ -87,6 +95,29 @@
     end
   end
 
+  context "paperclip" do
+    context "without cloudfront" do
+      it { should have_attached_file(:icon) }
+      it { should validate_attachment_content_type(:icon).
```
Comment (scarfdestroyer) [inline spec/models/organization_spec.rb:101]: Nice try hound, if i change this you will say the line is too long.

**Comment by joshsmith** [comment]
You'll want to merge `development` into this once #210 is merged. And you'll absolutely need to rewrite some specs and files here to take advantage of the new `Base64ImageDecoder`.

**Comment by scarfdestroyer** [comment]
Closes #207

**Comment by joshsmith** [comment]
@scarfdestroyer to clarify, we do not currently have a PUT/PATCH for these, right?

File: spec/models/organization_spec.rb [hunk 2]
```diff
@@ -87,6 +95,29 @@
...
+        allowing("image/png", "image/gif", "image/jpeg").
+        rejecting("text/plain", "text/xml")
+      }
+    end
+
+    context "with cloudfront" do
+
```
Comment (joshsmith) [inline spec/models/organization_spec.rb]: @scarfdestroyer definitely fix this one.

File: spec/requests/api/organizations_spec.rb [hunk 3]
```diff
@@ -48,6 +48,36 @@
...
+
+              expect(last_response.status).to eq 200
+
+              organization = Organization.last
+
+              expect(organization.base64_icon_data).to be_nil
+              expect(organization.icon.path).to_not be_nil
+              expect(organization.name).to eq "Test"
+              # expect icon saved from create action to be identical to our test photor
+              organization_icon_file = File.open("#{Rails.root}/spec/sample_data/default-avatar.png", 'r')
```
Comment (joshsmith) [inline spec/requests/api/organizations_spec.rb]: Also fix those single quotes.

File: spec/requests/api/organizations_spec.rb [hunk 4]
```diff
@@ -48,6 +48,36 @@
...
+              base64_saved_image = Base64.encode64(open(organization_icon_file) { |io| io.read })
```
Comment (joshsmith) [inline spec/requests/api/organizations_spec.rb]: And this.

File: spec/workers/add_organization_icon_worker_spec.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    it "doesn't touch icon" do
+      AddOrganizationIconWorker.new.perform(@organization.id)
+
+      @organization.reload
+      expect(@organization.icon.to_s).to eq "/icons/original/missing.png"
+    end
+  end
+end
```
Comment (joshsmith) [inline spec/workers/add_organization_icon_worker_spec.rb]: Add this.

File: app/workers/add_organization_icon_worker.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  include Sidekiq::Worker
+
+  def perform(organization_id)
+    organization = Organization.find(organization_id)
+    return unless organization.base64_icon_data
+    organization.icon = Base64ImageDecoder.decode(organization.base64_icon_data)
+    organization.base64_icon_data = nil
+    organization.save
+  end
+end
```
Comment (joshsmith) [inline app/workers/add_organization_icon_worker.rb]: Add this.

**Comment by joshsmith** [comment]
Everything looks good. Good work. Just fix those Hound issues I identified and we'll be good to merge.

**Comment by scarfdestroyer** [comment]
I think I fixed them right? I added the new lines.

**Pull request merged**